students/16th/team2/jmyoon/westagram/user/views.py

```diff
@@ -1,5 +1,74 @@
 import json
+import re
+import bcrypt
+import jwt
 
-from django.http import JsonResponse
-from django.views import View
+from django.http   import JsonResponse
+from django.views  import View
 
+from user.models   import User
+from my_settings   import SECRET_KEY
+
+class SignupView(View):
+    def post(self, request):
+        data = json.loads(request.body)
+
+        # email 정규표현식
+        vali_email = '^[a-zA-Z0-9+-_.]+@[a-zA-Z0-9-]+\.[a-zA-Z0-9-.]+$'
+        validate_email=re.compile(vali_email)
+
+        try:
+            if not data['email']:
+                return JsonResponse(
+                    {'MESSAGE' : 'KeyERROR'}, status = 400)
+            
+            if not validate_email.match(data['email']):
+                return JsonResponse(
+                    {'MESSAGE' : '잘못된 email형식'}, status = 400)
+
+            if len(data['password']) < 8 :
+                return JsonResponse(
+                    {'MESSAGE' : 'password는 8자리 이상'}, status = 400)
+
+            if (User.objects.filter(email = data['email']).exists()):
+                return JsonResponse(
+                    {'MESSAGE' : '이미 사용중'}, status = 400)
+
+            else :
+                pw = data['password'].encode()
+                pw_encrypt = bcrypt.hashpw(pw, bcrypt.gensalt())
+                pw_encrypt = pw_encrypt.decode()
```

Comment:
암호화 과정을 상세하게 스텝스텝 잘 적용해주셨네요 👍 하지만 조금 줄여서 더 깔끔하게 만들 수 있어보이네요!
```suggestion
                pw = data['password'].encode()
                hashed_password = bcrypt.hashpw(pw, bcrypt.gensalt()).decode()
```